### ISSUE TITLE
accessibility issue: alt text as captions doesn't work for long captions

### ISSUE THREAD

**Issue by vnbcs** [AprilSylph/XKit-Rewritten#476]
### Platform

macOS 12.0.1

### Browser

Firefox 97.0 (64-bit)

### Addon version

0.15.0

### Details

I use the accessibility feature that displays alt text as captions for images. When images don't have alt text, the word "image" is displayed. While looking through old posts, I saw "image" as the caption for two images that I knew I wrote alt text for.

<img width="562" alt="Screen Shot 2022-02-11 at 11 38 02 AM" src="https://user-images.githubusercontent.com/46567612/153631410-0f04e085-5b27-4c81-a21d-79ca404f0c07.png">

When I click into the image, the alt text is displayed as a caption.

<img width="1280" alt="Screen Shot 2022-02-11 at 11 38 49 AM" src="https://user-images.githubusercontent.com/46567612/153631542-ec92fcdb-5d9e-4177-9105-3e5523cf1380.png">

I'm guessing this is happening because my alt text was too long. I don't have an issue with having to click into an image to see the alt text, but if I only see the word "image" then I don't know the alt text is there at all. Perhaps adding the first couple words and then "..." might be better? I'm only mildly visually impaired, so it might be better to get feedback from other blind/low vision folks too.

Thanks so much!

**Comment by AprilSylph** [comment]
This isn't actually alt text! Legacy photosets let you set "captions" which only display in the lightbox, like in your second screenshot. I thought Tumblr had dropped support for them outside of themes, but I guess they were added back.

The AccessKit option is only for images with alt text set via the mobile and beta editors' options to set image descriptions; we don't touch the photoset lightbox at all.

You can report the issue with photoset captions to [Tumblr support](https://www.tumblr.com/support). Be sure to include that latter screenshot!

**Comment by AprilSylph** [comment]
Oh, and to be clear: the captions added by AccessKit which you see inside the post (not in the lightbox) are what screenreaders will announce. Setting photoset captions won't provide any accessibility benefits to users who rely on them.

**Comment by vnbcs** [comment]
Gotcha! I set those using the "image description" function on desktop. I've never seen a caption option, but I've only been on tumblr for about two years. I'm guessing there was a glitch at the time I made the post. Thank you!

**Comment by AprilSylph** [comment]
Tumblr still supports adding captions to images in photosets if you're not using the beta editor:

![image](https://user-images.githubusercontent.com/28949509/153641630-4ad8f985-8f8c-4180-9b95-392bd6ce32a4.png)

And naturally, if it was an older post, this would have been the only option at the time. But again, these captions *only* display in the lightbox, and provide no accessibility benefit.
